test/integration/router_test.go

```diff
@@ -457,16 +486,18 @@ func TestRouterPathSpecificity(t *testing.T) {
 	fakeMasterAndPod.EndpointChannel <- eventString(endpointEvent)
 	fakeMasterAndPod.RouteChannel <- eventString(routeEvent)
 
-	//ensure you can curl path but not main host
-	if err := waitForRoute(routeTestAddress, "www.example.com", "http", nil, tr.HelloPodPath); err != nil {
-		t.Fatalf("unexpected response: %q", err)
-	}
-	if _, err := getRoute(routeAddress, "www.example.com", "http", nil, ""); err != ErrUnavailable {
-		t.Fatalf("unexpected response: %q", err)
-	}
-	//ensure you can curl path with port in Host header
-	if err := waitForRoute(routeTestAddress, "www.example.com:80", "http", nil, tr.HelloPodPath); err != nil {
-		t.Fatalf("unexpected response: %q", err)
+	for _, proto := range protocols {
+		//ensure you can curl path but not main host
+		if err := waitForRoute(routeTestAddress, "www.example.com", proto.name, nil, tr.HelloPodPath); err != nil {
+			t.Fatalf("unexpected response: %q", err)
```

Comment:
Can you please also add which `protocol` specific test failed in the error messages.